analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -486,26 +489,31 @@ class SpdxDocumentFile(
         // For direct callers of this function mapDefinitionFiles() did not populate the map before, so add a fallback.
         val spdxDocument = spdxDocumentForFile.getOrPut(definitionFile) { SpdxModelMapper.read(definitionFile) }
 
-        // Guard against direct callers passing SPDX documents that do not describe a project.
-        val projectPackage = requireNotNull(spdxDocument.projectPackage()) {
-            "The SPDX document file at '$definitionFile' does not describe a project."
-        }
+        val projectPackage: SpdxPackage
```

Comment:
Nit: This `val` definition without an assignment looks a bit untypical. Maybe it can be assigned via the the `if`-`else` statement:
```
val projectPackage = if (!spdxDocument.isProject()) spdxDocument.packages[0]
else {
  ...
}

```
It may also help to extract the longer `else` block to a function.